### PR TITLE
Remove deprecated prometheus metrics from MS

### PR DESCRIPTION
**Pull request by JorTurFer**
### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Related to https://github.com/kedacore/keda/issues/3930

### CODE REVIEW

File: keda/README.md [hunk 1]
```diff
@@ -146,9 +146,8 @@ their default values.
 | `service.portHttps`                                        | HTTPS port for KEDA Metric Server service | `443` |
 | `service.portHttpsTarget`                                  | HTTPS port for KEDA Metric Server container | `6443` |
 | `prometheus.metricServer.enabled`                          | Enable metric server Prometheus metrics expose | `false` |
-| `prometheus.metricServer.port`                             | HTTP port used for exposing metrics server prometheus metrics | `9022` |
+| `prometheus.metricServer.port`                             | HTTP port used for exposing metrics server prometheus metrics | `8088` |
 | `prometheus.metricServer.portName`                         | HTTP port name for exposing metrics server prometheus metrics | `metrics` |
-| `prometheus.metricServer.path`                             | Path used for exposing metric server prometheus metrics | `/metrics` |
 | `prometheus.metricServer.podMonitor.enabled`               | Enable monitoring for metric server using podMonitor crd (prometheus operator) | `false` |
```
Comment (tomkerkhove) [inline keda/README.md:151]: Why do we still have all of these? Just for regular monitoring or?

File: keda/templates/22-metrics-deployment.yaml [hunk 2]
```diff
@@ -107,18 +107,14 @@ spec:
             {{- end }}
           args:
           - /usr/local/bin/keda-adapter
-          - --port={{ .Values.service.portHttpTarget }}
+          - --port={{ .Values.prometheus.metricServer.port }}
```
Comment (tomkerkhove) [inline keda/templates/22-metrics-deployment.yaml:110]: I don't think this is intentional?

Comment (JorTurFer) [reply keda/templates/22-metrics-deployment.yaml:110]: Yes it is. We had 2 different prometheus server and both allowed to change the port, one using `portHttpTarget` and the other with `metricServer.port`. As I had to remove one of them, I thought that `metricServer.port` is more clear about what does than `service.portHttpTarget`, but I have to remove `service.portHttpTarget` from values and readme (I missed that thing)

File: keda/README.md (hunk 1 above)
Comment (JorTurFer) [reply keda/README.md:151]: Yeap, the runtime's prometheus server is still there, so I updated the monitoring stuff to scrape those metrics. If we add some extra metrics in the future to the metrics server, we already have it monitored

File: keda/templates/22-metrics-deployment.yaml (hunk 2 above)
Comment (tomkerkhove) [reply keda/templates/22-metrics-deployment.yaml:110]: After reviewing the docs later on I realized we still have the default metrics there that are not KEDA specific so it's OK!

**Comment by EppO** [comment]
keda-operator is also exposing metrics on default port 8080 like keda_scaler_*, keda_scaled_objects_*, keda_resource_* or keda_trigger_*. There are no annotations for that currently, we would need to add: 
```
    prometheus.io/scrape: "true"
    prometheus.io/port: {{ .Values.prometheus.operator.port | quote }}
    prometheus.io/path: "/metrics"
```

in https://github.com/kedacore/charts/tree/main/keda/templates/manager/deployment.yaml
I also noticed that https://github.com/kedacore/charts/blob/main/keda/templates/manager/deployment.yaml#LL101C32-L101C32 is static, so if you change the value of prometheus.operator.port in the values.yaml, it may not work

**Comment by JorTurFer** [comment]
nice catches!
Thanks for the feedback ❤️  
I have to redo this PR soon, so I'll address it 😄

**Comment by zroubalik** [comment]
@JorTurFer could you please rebase this PR?

**Comment by JorTurFer** [comment]
> @JorTurFer could you please rebase this PR?

Yes, I'll do it during this week

**Comment by JorTurFer** [comment]
I still got this pending but I'm out of the office :(

**Comment by JorTurFer** [comment]
> @JorTurFer could you please rebase this PR?

done!

**Comment by JorTurFer** [comment]
PTAL @zroubalik , we need to merge this ASAP because the chart is "broken" until due to the parameters change in MS

**Review by zroubalik — APPROVED** [review]

**Comment by outime** [comment]
@JorTurFer @zroubalik @tomkerkhove my apologies for the multi-mention but I've just hit this and I was wondering if you're willing to push up a revision of the current KEDA chart to contain this and [few other commits](https://github.com/kedacore/charts/compare/v2.11.1...main)?

**Comment by JorTurFer** [comment]
> @JorTurFer @zroubalik @tomkerkhove my apologies for the multi-mention but I've just hit this and I was wondering if you're willing to push up a revision of the current KEDA chart to contain this and [few other commits](https://github.com/kedacore/charts/compare/v2.11.1...main)?

No worries for the mention 😄 
We plan to cut a release next week: https://github.com/kedacore/keda/issues/4795

**Pull request merged**